src/GslibIO.jl

```diff
@@ -52,28 +80,51 @@ Optionally set the value used for missings `na`.
 """
 function load_legacy(filename::AbstractString, dims::NTuple{3,Int};
                      origin=(0.,0.,0.), spacing=(1.,1.,1.), na=-999)
-  open(filename) do fs
-    # skip header
-    readline(fs)
-
-    # number of properties
-    nvars = parse.(Int, readline(fs))
+  dataspec = parse_legacy(filename)                     
 
-    # properties names
-    vars = [Symbol(readline(fs)) for i in 1:nvars]
+  # handle missing values
+  replace!(dataspec.data, na=>NaN)
 
-    # read property values
-    X = readdlm(fs)
+  # create data dictionary
+  table = (; zip(dataspec.varnames, eachcol(dataspec.data))...)
+  domain = RegularGrid(dims, origin, spacing)
 
-    # handle missing values
-    replace!(X, na=>NaN)
+  georef(table, domain)
+end
 
-    # create data dictionary
-    data = (; zip(vars, eachcol(X))...)
-    domain = RegularGrid(dims, origin, spacing)
+"""
+    load_legacy(filename, coordnames, na=-999)
 
-    georef(data, domain)
+Load legacy GSLIB `filename` into a PointSet using the properties in `coordnames` as coordinates.
+Optionally set the value used for missings `na`.
+"""
+function load_legacy(filename::AbstractString, coordnames=(:x, :y, :z); na=-999)
+  dataspec = parse_legacy(filename)
+
+  # handle missing values
+  replace!(dataspec.data, na=>NaN)
+
+  # we need to identify and separate coordinates and actual attributes 
+  # from the parsed variables
+
+  # find the position of each `coordnames` in `dataspec.varnames`
+  coordpos = []
+  for (i, coordname) in enumerate(coordnames)
+    pos = findall(x -> x == coordname, dataspec.varnames)
+    if isempty(pos)
+      @error "The coordinate name '$coordname' could not be found in the file"
+    else
+      append!(coordpos, pos)
+    end
   end
+  coords = transpose(dataspec.data[:, coordpos])
+
+  # create table with varnames not in coordnames
+  attrpos = [i for i in 1:length(dataspec.varnames) if i ∉ coordpos]
```

Comment:
Rewrite as `attrinds = setdiff(1:length(spec.varnames), coordinds)`

Reply:
Oh! I know now a new built-in function in the Julia language that does exactly what I wrote 👍🏽

Reply:
Maybe your version with the list comprehension is safer though, as I am not sure the `setdiff` preserves the order.

Reply:
Oh it preserves order in arrays, it is in the docstring 👍🏽

Reply:
I checked also and it preserves the order, but delete any duplicated element. Fortunately a range does not have duplicates!